public/app/features/dashboard-scene/scene/DashboardControls.tsx

```diff
@@ -74,6 +75,20 @@ export class DashboardControls extends SceneObjectBase<DashboardControlsState> {
       refreshPicker: state.refreshPicker ?? new SceneRefreshPicker({}),
       ...state,
     });
+
+    this.addActivationHandler(() => {
+      let refreshPickerDeactivation: CancelActivationHandler | undefined;
+
+      if (this.state.hideTimeControls) {
+        refreshPickerDeactivation = this.state.refreshPicker.activate();
+      }
+
+      return () => {
+        if (refreshPickerDeactivation) {
+          refreshPickerDeactivation();
+        }
+      };
+    });
   }
```

Comment:
@dprokop we could also add a isHidden property to SceneRefreshPicker (so we can render it but it internally just render null when hidden)

Reply:
But is there a need for that in the library API? I am hesitant of introducing one-off property that will serve only core dashboard case.